IronFoundry.Container.Shared/ProcessRunner.cs

```diff
@@ -116,5 +120,29 @@ public IProcess Run(ProcessRunSpec runSpec)
         public void StopAll(bool kill)
         {
         }
+
+        /// <summary>
+        /// Returns default environment for the process.
+        /// If credentials are specified then the default environment is the default for that user.
+        /// Otherwise the default is to inherit from this process.
+        /// </summary>
+        private Dictionary<string, string> CreateDefaultProcessEnvironment(NetworkCredential credential)
+        {
+            EnvironmentBlock envBlock = new EnvironmentBlock();
+
+            if (credential == null)
+            {
+                envBlock = EnvironmentBlock.Create(Environment.GetEnvironmentVariables());
+            }
+            else
+            {
+                using (var safeUserToken = Utils.LogonAndGetUserPrimaryToken(credential))
+                {
+                    envBlock = EnvironmentBlock.CreateForUser(safeUserToken.DangerousGetHandle());
```

Comment:
Windows has no good way of getting the environment variables of a particular process.  The warden needs those so it can create the env.log file.  There are 2 ways I can think of to get the environment variables.  1. We always specify them when we create the process  2. We read the process memory.  

I went with option 1 which seemed less complicated.